djautotask/sync.py

```diff
@@ -879,6 +882,43 @@ def sync_related(self, instance):
 
         self.sync_children(*sync_classes)
 
+    # TODO move to parent sync when generalized, OR remove when
+    #  "sync/card packager" class created
+    def get_changed_values(self, record, **kwargs):
+        """
+        Only create tickets with manually selected fields, so AT can set its
+        own defaults.
+
+        """
+        changed_field_keys = kwargs.get('changed_fields')
+
+        new_record = {}
+        if changed_field_keys:
+            for field in changed_field_keys:
+                new_record[field] = getattr(record, field)
+
+        return new_record
+
+    def create(self, **kwargs):
+        """
+        Make a request to Autotask to create a Ticket.
+        """
+
+        if self.client.impersonation_resource:
+            description = kwargs['description']
+        else:
+            description = "{}\n\nTicket was created by {} {}.".format(
+                kwargs['description'],
```

Comment:
resource and description here don't refer to fields on the ticket right? They're here purely for tracking?

Reply:
resource is for tracking, but description is the field on the ticket.

Reply:
ok, my concern was what if you want to create a ticket without a resource or description. This seems to make it like a description is required to create a ticket. But maybe I'm misunderstanding it.

Reply:
A description isn't required, but if they are creating it and not impersonating we force adding it. Originally this wasn't going to be the "form commit false" pattern, and this is leftover from that. I need to update it so it gets the description from the record generated by the form (which I intended to always have a blank description).

Updated now, I have not tested it.

Reply:
I see. Thanks for clarifying. I'll try the change.